src/rnr_score_hv/innov_netcdf.py

```diff
@@ -0,0 +1,329 @@
+"""
+Copyright 2022 NOAA
+All rights reserved.
+
+Collection of methods to facilitate file/object retrieval
+
+"""
+from collections import namedtuple
+from dataclasses import dataclass, field
+from datetime import datetime
+import netCDF4
+
+from rnr_score_hv.config_base import ConfigInterface
+from rnr_score_hv import file_utils
+
+valid_metrics = [
+    'temperature',
+    'spechumid',
+    'uvwind'
+]
+
+VALID_STATS = [
+    'bias',
+    'count',
+    'rmsd'
+]
+
+
+@dataclass
+class Region:
+    ''' region object storing region name and min/max latitude bounds '''
+    name: str
+    min_lat: float
+    max_lat: float
+
+    def __post_init__(self):
+        if self.min_lat > self.max_lat:
+            msg = f'min_lat must be less than max_lat - min_lat: {self.min_lat}, '\
+                f'max_lat: {self.max_lat}'
+            raise ValueError(msg)
+        if self.max_lat < self.max_lat:
+            msg = f'min_lat must be greater than min_lat - min_lat: {self.min_lat}, '\
+                f'max_lat: {self.max_lat}'
+            raise ValueError(msg)
+
+        if abs(self.min_lat) > 90 or abs(self.max_lat) > 90:
+            msg = f'min_lat or max_lat is out of allowed range, must be greater' \
+                f' than -90 and let than 90 - min_lat: {self.min_lat}, ' \
+                f'max_lat: {self.max_lat}'
+            raise ValueError(msg)
+
+DEFAULT_REGIONS = [
+    Region('equatorial', -5, 5),
+    Region('global', -90, 90),
+    Region('north_hemis', 20, 60),
+    Region('tropics', -20, 20),
+    Region('south_hemis', -60, -20)
+]
+
+
+@dataclass
+class MetricMeta:
+    """
+    Information pertaining to where a particular measurement type
+    file can be found.
+
+    Parameters:
+    -----------
+    name: str - metric name (temperature, uvwind, or spechumid)
+    file_meta: dict - filename meta data containing file path
+               filename format, and cycle time
+    filename: str - configured full filename and absolute path
+    cycletime: datetime - for reference, the relevant cycle time
+               for the measurements
+    """
+    name: str
+    file_meta: dict
+    filename: str = field(default_factory=str, init=False)
+    cycletime: datetime = field(init=False)
+
+    def __post_init__(self):
+        try:
+            # build filename from the config meta data
+            file_path = self.file_meta.get('filepath')
+            cycle = self.file_meta.get('cycle')
+
+            self.cycletime = datetime.strptime(
```

Comment:
do we need a method to check if the cycle time is a valid time?

Reply:
That's a good idea, yeah.  I'll add one.

Reply:
When you say valid time, do you mean within the 30 year range we set up or simply a valid cycle like [00z, 06z, 12z, and 18z]?

Reply:
hm. it gets overly complicated. Like the month is between 1-12, day between 1-31, hour between 0:24
For year. not sure. That is composed of digits and not literals? 

NOt sure if this is important.